src/custom/pages/Profile/index.tsx

```diff
@@ -223,13 +227,24 @@ export default function Profile() {
             </span>
           </BalanceDisplay>
           <CardActions>
-            <ExtLink href={getBlockExplorerUrl(chainId || 1, COW_CONTRACT_ADDRESS[chainId || 1], 'address')}>
+            <ExtLink
+              title="View contract"
+              href={getBlockExplorerUrl(chainId || 1, COW_CONTRACT_ADDRESS[chainId || 1], 'address')}
+            >
               Contract ↗
             </ExtLink>
 
-            <AddToMetamask currency={COW[chainId || 1] as Currency | undefined} />
+            {library?.provider?.isMetaMask && (
+              <AddToMetamask shortLabel={true} currency={COW[chainId || 1] as Currency | undefined} />
```

Comment:
why do you use `as` here

Reply:
Without it:
<img width="812" alt="Screen Shot 2022-04-06 at 16 39 45" src="https://user-images.githubusercontent.com/31534717/162013792-89e91659-f1dd-4bc0-9ade-4db382d1fef7.png">

Reply:
that's a useful error, you can't pass undefined here. change the 247 conditional to check that currency is indeed currency

actually why do you even need `| undefined` in 248? it's the COW token (hardcoded) and you're always passing a network

Reply:
Updated now. Right now it checks if the const `currencyCOW` exists. I assume `COW[chainId || ChainId.MAINNET]` always returns the right type in this case (`Token`). But how best to add a type guard here? (tried `typeof` but can't use it, and `instance of` is not appropriate as not an instance of a Class?).

Reply:
<img width="790" alt="image" src="https://user-images.githubusercontent.com/21335563/162030619-fe141afd-8531-420c-94a7-46fef396f095.png">

it doesn't need to check COW, it knows it exists. do you have the error without the condiational check?

Reply:
No error, when I remove the `currencyCOW` check. Removed it now. Initially I understood your comment to make sure we first check/make sure that `currencyCOW` matches the `Currency` type by doing a type check. But I guess not needed in this case.